src/App/Fossa/Analyze/Upload.hs

```diff
@@ -211,3 +216,24 @@ buildProjectSummary project locator projectUrl = do
       , "url" .= projectUrl
       , "id" .= renderLocator locator
       ]
+
+checkUploadResponseForWarnings ::
```

Comment:
[nit] This function does more than its name implies: it _emits_ the returned warnings into the debug bundle. Consider a name like `emitBackendWarningsToBundle` instead.